src/codons.py

```diff
@@ -60,8 +67,16 @@ def translate_codons(codons: list[str]) -> list[str]:
     True
 
     """
-    # FIXME: Implement the function
-    return []
+    j = 0
+    while j < len(codons):
+        if len(codons[j]) % 3 != 0:
+            return None
+        if codons[j].upper() not in CODON_MAP.keys():
```

Comment:
You should ask

```python
if codons[j].upper() not in CODONS_MAP:
    ...
```

rather than

```python
if codons[j].upper() not in CODONS_MAP.keys():
    ...
```

Explicitly pulling out the keys make it less efficient (for reasons we see towards the end of the class), and it just looks messy to a Python programmer's eyes :)